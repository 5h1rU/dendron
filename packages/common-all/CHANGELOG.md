# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.3.11](https://github.com/dendronhq/dendron/compare/v0.3.10...v0.3.11) (2020-07-13)

**Note:** Version bump only for package @dendronhq/common-all

## [0.3.10](https://github.com/dendronhq/dendron/compare/v0.1.6...v0.3.10) (2020-07-13)

### Features

- auto add nodes when deleted or created outside of dendron ([8c311bd](https://github.com/dendronhq/dendron/commit/8c311bda948a1d54088c49fd70eb65d24af5d68f))
- better schema suggestions ([03656bc](https://github.com/dendronhq/dendron/commit/03656bc007810457cb6846f0d6adacab4a7fbd3a))
- match namespace schemas ([7a67b8b](https://github.com/dendronhq/dendron/commit/7a67b8b2fb7caa1b97ee6d492d2801782abecdf6))
- show node descriptions ([aca86f2](https://github.com/dendronhq/dendron/commit/aca86f2a5fd6ee481f93553693a098db0e322890))
- surface unknown schemas ([d014965](https://github.com/dendronhq/dendron/commit/d0149652c985c69a4b2607984d578902820077f1))
- updated icons for schemas ([21804eb](https://github.com/dendronhq/dendron/commit/21804eba61c8dd49e499edd5d548d9d601224e8e))

## [0.3.9](https://github.com/dendronhq/dendron/compare/v0.3.8...v0.3.9) (2020-07-09)

### Features

- updated icons for schemas ([b7a2d8a](https://github.com/dendronhq/dendron/commit/b7a2d8aa517cf88d7a93d07cd2ef19305e48d069))

## [0.3.7](https://github.com/dendronhq/dendron/compare/v0.3.6...v0.3.7) (2020-07-08)

### Features

- better schema suggestions ([ad74bc0](https://github.com/dendronhq/dendron/commit/ad74bc009e1544319a49689394ab8d6b684f6578))
- show node descriptions ([e08fce9](https://github.com/dendronhq/dendron/commit/e08fce994153e28fe504b85e6d9bc1f5fdd93e20))

## [0.3.6](https://github.com/dendronhq/dendron/compare/v0.3.5...v0.3.6) (2020-07-07)

### Features

- auto add nodes when deleted or created outside of dendron ([a7e1ac9](https://github.com/dendronhq/dendron/commit/a7e1ac9b8a4f7f0592ab1b9f86a7a40182693a73))

# [0.3.0](https://github.com/dendronhq/dendron/compare/v0.2.20...v0.3.0) (2020-07-05)

### Features

- match namespace schemas ([1a960bf](https://github.com/dendronhq/dendron/commit/1a960bf26f8984e541b3eb118f60bdc09d8250fe))
- surface unknown schemas ([9bf4d0e](https://github.com/dendronhq/dendron/commit/9bf4d0e61cce2f76bddae1f686f29474201466cb))
