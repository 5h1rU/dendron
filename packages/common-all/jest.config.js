const jestConfig = require("../../jest.config");

console.log(jestConfig);
module.exports = {
  ...jestConfig
};
