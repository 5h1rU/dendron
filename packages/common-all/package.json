{
  "name": "@dendronhq/common-all",
  "version": "0.1.6",
  "description": "common-all",
  "license": "MIT",
  "repository": "https://github.com/dendronhq/dendron",
  "author": {
    "name": "Kevin Lin",
    "email": "kevin@thence.io",
    "url": "https://kevinslin.com",
    "organization": true
  },
  "keywords": [
    "common"
  ],
  "files": [
    "lib"
  ],
  "main": "lib/index",
  "types": "lib/index",
  "scripts": {
    "clean": "rimraf lib && rimraf coverage",
    "format": "",
    "lint": "",
    "prepublishOnly": "npm run build",
    "prebuild": "npm run clean && npm run format && npm run lint && echo Using TypeScript && tsc --version",
    "build": "tsc --pretty",
    "test": "jest",
    "coverage": "jest --coverage",
    "watch": "npm run build -- --watch",
    "test:watch": "jest --watch"
  },
  "dependencies": {
    "dropbox": "^4.0.30",
    "isomorphic-fetch": "^2.2.1",
    "lodash": "^4.17.15",
    "pino": "^6.3.1",
    "uuid": "^8.1.0",
    "yamljs": "^0.3.0"
  },
  "devDependencies": {
    "@types/jest": "^23.3.3",
    "@types/lodash": "^4.14.152",
    "@types/node": "^10.11.4",
    "@types/pino": "^6.0.1",
    "@types/uuid": "^8.0.0",
    "@types/yamljs": "^0.2.31",
    "coveralls": "^3.0.2",
    "jest": "^23.6.0",
    "prettier": "^1.14.3",
    "rimraf": "^2.6.2",
    "ts-jest": "^23.10.3",
    "ts-node": "^7.0.1",
    "tslint": "^5.11.0",
    "tslint-config-prettier": "^1.15.0",
    "typescript": "^3.1.1"
  },
  "engines": {
    "node": ">=10.0.0"
  },
  "jest": {
    "preset": "ts-jest"
  },
  "gitHead": "403eb012f8b34bc115dd7dd86f80c76b39329f3b"
}
