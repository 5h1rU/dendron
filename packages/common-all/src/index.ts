export * from "./types";
export * from "./store";
export * from "./node";
export * from "./helpers";
