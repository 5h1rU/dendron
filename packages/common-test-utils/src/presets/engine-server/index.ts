import { ENGINE_DELETE_PRESETS } from "./delete";
import ENGINE_MULTI_TEST_PRESET from "./engine-multi";
import ENGINE_SINGLE_TEST_PRESET from "./engine-single";
import { ENGINE_GET_NOTE_BY_PATH_PRESETS } from "./getByPath";
import { ENGINE_INIT_PRESETS } from "./init";
import NOTE_REF from "./note-refs";
import { ENGINE_QUERY_PRESETS } from "./query";
import { ENGINE_RENAME_PRESETS } from "./rename";
import { ENGINE_UPDATE_PRESETS } from "./update";
import { ENGINE_WRITE_PRESETS } from "./write";

export const ENGINE_SERVER = {
  NOTE_REF,
  ENGINE_MULTI_TEST_PRESET,
  ENGINE_SINGLE_TEST_PRESET,
  ENGINE_WRITE_PRESETS,
  ENGINE_INIT_PRESETS,
  ENGINE_UPDATE_PRESETS,
  ENGINE_DELETE_PRESETS,
  ENGINE_GET_NOTE_BY_PATH_PRESETS,
  ENGINE_RENAME_PRESETS,
  ENGINE_QUERY_PRESETS,
};

export const ENGINE_PRESETS = [
  // {name: "init", presets: ENGINE_SERVER.ENGINE_INIT_PRESETS},
  // {name: "delete", presets: ENGINE_SERVER.ENGINE_DELETE_PRESETS},
  // {name: "getByPath", presets: ENGINE_SERVER.ENGINE_GET_NOTE_BY_PATH_PRESETS},
  { name: "query", presets: ENGINE_SERVER.ENGINE_QUERY_PRESETS },
  // {name: "rename", presets: ENGINE_SERVER.ENGINE_RENAME_PRESETS},
  // {name: "update", presets: ENGINE_SERVER.ENGINE_UPDATE_PRESETS},
  // {name: "write", presets: ENGINE_SERVER.ENGINE_WRITE_PRESETS},
];
