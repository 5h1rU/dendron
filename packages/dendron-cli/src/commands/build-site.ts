import {
  DendronSiteConfig,
  DEngine,
  DNodeUtils,
  Note,
} from "@dendronhq/common-all";
import { resolvePath } from "@dendronhq/common-server";
import fs from "fs-extra";
import matter from "gray-matter";
import _ from "lodash";
import path from "path";
import { BaseCommand } from "./base";

type CommandOpts = {
  engine: DEngine;
  config: DendronSiteConfig;
} & CommandCommonOpts;

type CommandCommonOpts = {
  dendronRoot: string;
};

type CommandOutput = void;

type DendronJekyllProps = {
  hpath: string;
  permalink?: string;
};

function stripSiteOnlyTags(note: Note) {
  const re = new RegExp(/(?<raw><!--SITE_ONLY(?<body>.*)-->)/, "ms");
  let matches;
  let doc = note.body;
  do {
    matches = doc.match(re);
    if (matches) {
      // @ts-ignore
      const { raw, body } = matches.groups;
      doc = doc.replace(raw, body);
    }
  } while (matches);
  return doc;
}

function wikiLinkToMd(
  note: Note,
  engine: DEngine,
  opts?: { linkPrefix: string }
) {
  let matches;
  let doc = note.body;
  do {
    matches = doc.match(/(?<raw>\[\[(?<link>[^\]]+)\]\])/);
    if (matches) {
      // @ts-ignore
      const { raw, link } = matches.groups;
      const [first, rest] = link.split("|");
      let title: string | undefined;
      let mdLink: string;
      // we have a piped title
      if (rest) {
        title = _.trim(first);
        mdLink = _.trim(rest);
      } else {
        mdLink = _.trim(first);
      }
      const noteFromLink = _.find(engine.notes, { fname: mdLink });
      if (!noteFromLink) {
        throw Error(`${mdLink} not found. file: ${note.fname}`);
      }
      if (!title) {
        title = _.trim(noteFromLink.title);
      }
      let noteLink = noteFromLink.id;
      if (opts?.linkPrefix) {
        noteLink = `${opts.linkPrefix}/${noteLink}`;
      }
      const newLink = `[${title}](${noteLink})`;
      doc = doc.replace(raw, newLink);
    }
  } while (matches);
  return doc;
}

// @ts-ignore
function imageLinkConverter(note: Note) {
  let matches;
  let doc = note.body;
  do {
    matches = doc.match(/(\(?<link>assets\/images[^)]+\))/);
    if (matches) {
      // @ts-ignore
      const { link } = matches.groups;
      const linkReplace = link.replace("(assets", "(/assets");
      doc = doc.replace(link, linkReplace);
    }
  } while (matches);
  return doc;
}

function note2JekyllMdFile(
  note: Note,
  opts: { notesDir: string; engine: DEngine } & DendronSiteConfig
) {
  const meta = DNodeUtils.getMeta(note, {
    pullCustomUp: true,
    ignoreNullParent: true,
  });
  const jekyllProps: DendronJekyllProps = {
    hpath: note.path,
  };
  let linkPrefix = "";
  if (opts.noteRoot === meta.fname) {
    jekyllProps["permalink"] = "/";
    linkPrefix = path.basename(opts.notesDir);
  }
  // pull children of root to the top
  if (note.parent?.fname === opts.noteRoot) {
    delete meta["parent"];
  }
  // delete parent from root
  note.body = stripSiteOnlyTags(note);
  // TODO: HACK
  if (note.id !== "f1af56bb-db27-47ae-8406-61a98de6c78c") {
    note.body = wikiLinkToMd(note, opts.engine, { linkPrefix });
  }
  const filePath = path.join(opts.notesDir, meta.id + ".md");
  return fs.writeFile(
    filePath,
    matter.stringify(note.body || "", { ...meta, ...jekyllProps })
  );
}

export class BuildSiteCommand extends BaseCommand<CommandOpts, CommandOutput> {
  async execute(opts: CommandOpts) {
    const { engine, config, dendronRoot } = _.defaults(opts, {});
    const { siteRoot: siteRootRaw, noteRoot } = config;

    const siteRoot = resolvePath(siteRootRaw, dendronRoot);
    const siteNotesDir = "notes";
    const siteNotesDirPath = path.join(siteRoot, siteNotesDir);
    const L = this.L;
    // ({
    //   ctx: "BuildSiteComman",
    //   siteRoot,
    //   dendronRoot,
    //   noteRoot
    // });
    L.info({ msg: "enter", siteNotesDirPath });
    fs.ensureDirSync(siteNotesDirPath);
    fs.emptyDirSync(siteNotesDirPath);

    const root: Note = _.find(engine.notes, { fname: noteRoot }) as Note;
    if (_.isUndefined(root)) {
      throw Error(`root ${root} not found`);
    }
    const nodes: Note[] = [root];
    const out = [];

    // delete parent from the root
    root["parent"] = null;
    root.custom.nav_order = 0;
    root.title = _.capitalize(root.title);

    while (!_.isEmpty(nodes)) {
      const node = nodes.pop() as Note;
      out.push(
        note2JekyllMdFile(node, {
          notesDir: siteNotesDirPath,
          engine,
          ...config,
        })
      );
      node.children.forEach((n) => nodes.push(n as Note));
    }

    // TODO: need to rewrite links before this is ready
    const assetsDir = "assets";
    // notes/assets
    //const noteAssetsDir = path.join(siteNotesDirPath, assetsDir);
    const vaultAssetsDir = path.join(engine.props.root, assetsDir);
    // docs/assets
    const siteAssetsDir = path.join(siteRoot, assetsDir);

    const copyP = new Promise((resolve, reject) => {
      fs.copy(
        path.join(vaultAssetsDir, "images"),
        path.join(siteAssetsDir, "images"),
        (err) => {
          if (err) {
            err.message += JSON.stringify({ vaultAssetsDir, siteAssetsDir });
            reject(err);
          }
          L.info({ msg: "finish copying" });
          resolve();
        }
      );
    });
    await Promise.all(nodes);
    await copyP;
    L.info({ msg: "exit" });
    return;
  }
}

export type BuildSiteCliOpts = {
  vault: string;
} & CommandCommonOpts;

// rm -r site-builder/docs 2>/dev/null || true
// mkdir site-builder/docs
// echo "copying files..."
// cp vault/* site-builder/docs
