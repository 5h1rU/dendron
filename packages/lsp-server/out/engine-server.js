"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DendronEngineServer = void 0;
const engine_server_1 = require("@dendronhq/engine-server");
class DendronEngineServer {
    constructor(opts) {
        this.opts = opts;
    }
    static getOrCreate(opts) {
        if (!DendronEngineServer._instance) {
            DendronEngineServer._instance = new DendronEngineServer(opts);
        }
        return DendronEngineServer._instance;
    }
    init() {
        const root = this.opts.roots[0];
        if (!root) {
            return;
        }
        const engine = engine_server_1.DendronEngine.getOrCreateEngine({
            root: root.uri,
            forceNew: true,
        });
        this._engine = engine;
        this._engine.init();
    }
}
exports.DendronEngineServer = DendronEngineServer;
//# sourceMappingURL=engine-server.js.map