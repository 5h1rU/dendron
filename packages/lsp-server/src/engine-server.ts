import { WorkspaceFolder } from "vscode-languageserver";
import { DendronEngine } from "@dendronhq/engine-server";

type DendronEngineServerOpts = {
  roots: WorkspaceFolder[];
};

export class DendronEngineServer {
  protected _engine?: DendronEngine;

  constructor(public opts: DendronEngineServerOpts) {}

  static _instance: DendronEngineServer | null;

  static getOrCreate(opts: DendronEngineServerOpts) {
    if (!DendronEngineServer._instance) {
      DendronEngineServer._instance = new DendronEngineServer(opts);
    }
    return DendronEngineServer._instance;
  }

  init() {
    const root = this.opts.roots[0];
    if (!root) {
      return;
    }
    const engine = DendronEngine.getOrCreateEngine({
      root: root.uri,
      forceNew: true,
    }) as DendronEngine;
    this._engine = engine;
    this._engine.init();
  }
}
