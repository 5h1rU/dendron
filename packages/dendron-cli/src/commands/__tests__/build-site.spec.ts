import { DEngine } from "@dendronhq/common-all";
import fs from "fs-extra";
import {
  LernaTestUtils,
  EngineTestUtils,
  FileTestUtils,
  readMD,
} from "@dendronhq/common-server";
import { DendronEngine } from "@dendronhq/engine-server";
import { BuildSiteCommand } from "../build-site";
import path from "path";

function setupTmpDendronDir() {
  return EngineTestUtils.setupStoreDir(
    LernaTestUtils.getFixturesDir("store"),
    FileTestUtils.tmpDir().name
  );
}

describe("build-site", () => {
  let root: string;
  let engine: DEngine;

  beforeEach(() => {
    root = setupTmpDendronDir();
    engine = DendronEngine.getOrCreateEngine({
      root,
      forceNew: true,
      mode: "exact",
    });
  });

  afterEach(() => {
    // expect(actualFiles).toEqual(expectedFiles);
    console.log("bond");
    console.log(root);
    // fs.removeSync(root);
  });
  test("basic", async () => {
    const siteRoot = FileTestUtils.tmpDir();
    await engine.init();
    const config = {
      noteRoot: "root",
      siteRoot: siteRoot.name,
    };
    const dendronRoot = root;
    await new BuildSiteCommand().execute({ engine, config, dendronRoot });
    const buildDir = path.join(siteRoot.name, "notes");
    const { data, content } = readMD(path.join(buildDir, "foo.md"));
    expect(content).toMatchSnapshot("bond-foo");
    expect(data.id).toEqual("foo");
    expect(content.indexOf("- [refactor-one](refactor.one)") >= 0).toBe(true);
    expect(1).toEqual(1);
  });
});
