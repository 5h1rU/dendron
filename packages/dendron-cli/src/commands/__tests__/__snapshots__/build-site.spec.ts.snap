// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`build-site basic: bond-foo 1`] = `
"
- [refactor-one](refactor.one)
"
`;
